internal/charmstore/resources.go

```diff
@@ -0,0 +1,255 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmstore // import "gopkg.in/juju/charmstore.v5-unstable/internal/charmstore"
+
+import (
+	"io"
+	"time"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charm.v6-unstable"
+	"gopkg.in/juju/charmrepo.v2-unstable/csclient/params"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+
+	"gopkg.in/juju/charmstore.v5-unstable/internal/mongodoc"
+)
+
+var resourceNotFound = errgo.Newf("resource not found")
+
+// ListResources returns the list of resources for the charm at the
+// latest revision for each resource.
+func (s Store) ListResources(entity *mongodoc.Entity, channel params.Channel) ([]*mongodoc.Resource, error) {
```

Comment:
LatestResources() does exactly what you described.  For the given charm and channel, it gives the set of resources at the appropriate revisions.  For the unpublished channel it gives the latest revision of each resource.  For other channels it gives the resource revisions published with that charm/channel pair.  This *should* be what the client does as well.

What was confusing?  The doc comment?  I'll clarify it.